bin/genesis

```diff
@@ -758,7 +758,23 @@ sub __prompt_for_line {
 	# being an optional error message
 	my $validate;
 	if (defined($validation)) {
-		if ($validation =~ m/^\/(.*)\/(i?m?s?)$/) {
+		if ($validation eq "url") {
+			$validate = sub() {
+				if (is_valid_uri($_[0])) {
+					return "";
+				} else {
+					return ($_[1] ? $_[1] :"$_[0] is not a valid URL\n");
+				}
+			}
+		} elsif ($validation eq "port") {
+			$validate = sub() {
+				if ( ($_[0] =~ /^\s*[\+\-]?\d+\s*$/) && ($_[0] >= 0 ) && ($_[0] <= 65535) ) {
```

Comment:
You don't need to escape '+' or '-' inside of the character class brackets - they lose their special meaning as soon as you cross into `[...]` territory.

That being said, if this is to validate ports, why are you even allowing a sign on the number?  port "-80" isn't a real port.  I would drop the sign and just check against `\d+`

Reply:
Thanks, will modify